Add In|Not In Array Operation

lucasangi opened the pull request:
First of all, congratulations on the amazing library, it has helped me a lot! :clap: 

I opened this PR to allow the use of `in [foo, bar]` instead of multiple clauses connected with `or` or one clause with `regex`. 

NOTE: In my research I couldn't find a syntax pattern for this clause so I adopted the JS array pattern (`[]`).

Galbar commented on src/Galbar/JsonPath/Expression/InArray.php:
```diff
@@ -0,0 +1,30 @@
+    {
+        return trim(trim($listValue), '\'');
+    }
+}
```
Could you add an empty line at the end of the file?

lucasangi replied:
Done!

Galbar commented on src/Galbar/JsonPath/Language/Regex.php:
```diff
@@ -38,6 +38,7 @@ class Regex
     const EXPR_STRING = '/^(?:\'(.*)\'|"(.*)")$/';
     const EXPR_REGEX = '/^\/.*\/i?x?$/';
     const BINOP_COMP = '/^(.+)\s*(==|!=|<=|>=|<|>|=\~)\s*(.+)$/';
+    const BINOP_ARRAY = '/^(.+)\s*in\s*\[(.+)\]$/';
```
For ease of readability, could you rename the const to `BINOP_IN_ARRAY`?

lucasangi replied:
Done!

Galbar commented on src/Galbar/JsonPath/Expression/InArray.php:
```diff
@@ -0,0 +1,30 @@
+
+    private static function prepareListValue(string $listValue): string
+    {
+        return trim(trim($listValue), '\'');
```
Did you consider using `Value::evaluate(...)` to parse the possible values in the array so that it not only works with strings but with numbers, booleans, etc.? If you did and decided against it, I'd be curious to know the reason why.

lucasangi replied:
I hadn't considered this possibility, thanks for the suggestion! 

I changed the `prepareListValue` method to use the `Value::evaluate(...)` ([changes](https://github.com/Galbar/JsonPath-PHP/pull/65/files#diff-80947ea24e8df6e501519792f8d36cbd7e39962e39c95dd09e8b26aa83c3cbd0))).

Galbar reviewed (changes requested):
Hi, Thanks for the contribution! It looks great and I really like the idea.

Before I can merge this I'd like to discuss some details I've noticed on my first quick look at it. I've added most of them as comments in the PR. Here's two more:
- could you add a test case in `testParsingErrorsProvider` for a case when there are two `[` or `]`, one of which is not opening/closing any expression. Just to make sure that it fails in an expected way.
- could you update the readme file with an updated language specification including this changes?

Thanks again for the contribution! Let's resolve this comments and have it merged!

lucasangi commented:
> 

Thanks for the review! I apologize for the delay in the correction.

I added some test cases in `testParsingErrorsProvider` and some examples in the README file. 
Please check if these changes meet your observations.

lucasangi commented:
> Hi, Thanks for the contribution! It looks great and I really like the idea.
> 
> Before I can merge this I'd like to discuss some details I've noticed on my first quick look at it. I've added most of them as comments in the PR. Here's two more:
> 
>     * could you add a test case in `testParsingErrorsProvider` for a case when there are two `[` or `]`, one of which is not opening/closing any expression. Just to make sure that it fails in an expected way.
> 
>     * could you update the readme file with an updated language specification including this changes?
> 
> 
> Thanks again for the contribution! Let's resolve this comments and have it merged!

Thanks for the review! I apologize for the delay in the correction.

I added some test cases in `testParsingErrorsProvider` and some examples in the README file. Please check if these changes meet your observations.

Galbar reviewed (approved):
There are two small changes that I'll do to the README.md file, in a follow-up commit once this is merged. Everything else looks good

Galbar commented:
The tests failed because you seem to have used "modern" PHP and this library supports very old PHP versions (5.4). I've validated it locally. I'll merge this and upload a patch to adapt it to old PHP syntax.

Thanks a lot for the contribution!

Galbar commented:
I've published a new version containing this changes: https://github.com/Galbar/JsonPath-PHP/releases/tag/3.0
